tests/Unit/Elliptic/Systems/Xcts/Events/Test_ObserveAdmIntegrals.cpp

```diff
@@ -6,156 +6,245 @@
+    for (int a = 0; a < 4; a++) {
+      for (int b = 0; b <= a; b++) {
+        for (int c = 0; c < 4; c++) {
+          for (int d = 0; d < 4; d++) {
+            spacetime_metric.get(a, b) += boost_matrix.get(c, a) *
+                                          boost_matrix.get(d, b) *
+                                          barred_spacetime_metric.get(c, d);
+          }
+        }
+      }
+    }
+
+    // Do the 3+1 decomposition.
+    const auto spatial_metric = gr::spatial_metric(spacetime_metric);
     const auto& inv_spatial_metric =
-        get<gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Inertial>>(
-            background_fields);
-    const auto& extrinsic_curvature =
-        get<gr::Tags::ExtrinsicCurvature<DataVector, 3, Frame::Inertial>>(
-            background_fields);
-    const auto& trace_extrinsic_curvature =
-        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(background_fields);
-
-    // Compute face normal (related to the conformal metric)
+        determinant_and_inverse(spatial_metric).second;
+    const auto shift = gr::shift(spacetime_metric, inv_spatial_metric);
+    const auto lapse = gr::lapse(shift, spacetime_metric);
+
+    // Do the conformal decomposition.
+    // Note that we choose the same conformal factor as the one used for the
+    // Schwarzschild solution in isotropic coordinates. Other conformal factors
+    // could also work.
+    const auto solution_conformal_factor = solution.variables(inertial_coords,
+        tmpl::list<Xcts::Tags::ConformalFactor<DataVector>>{});
+    const auto& conformal_factor =
+        get<Xcts::Tags::ConformalFactor<DataVector>>(solution_conformal_factor);
+    const auto conformal_metric = tenex::evaluate<ti::i, ti::j>(
+        spatial_metric(ti::i, ti::j) / pow<4>(conformal_factor()));
+    const auto inv_conformal_metric = tenex::evaluate<ti::I, ti::J>(
+        inv_spatial_metric(ti::I, ti::J) * pow<4>(conformal_factor()));
+
+    // Compute spatial derivatives (needed for extrinsic curvature).
+    const auto deriv_lapse = partial_derivative(lapse, mesh, inv_jacobian);
+    const auto deriv_shift = partial_derivative(shift, mesh, inv_jacobian);
+    const auto deriv_spatial_metric =
+        partial_derivative(spatial_metric, mesh, inv_jacobian);
+    const auto deriv_conformal_factor =
+        partial_derivative(conformal_factor, mesh, inv_jacobian);
+    const auto deriv_conformal_metric =
+        partial_derivative(conformal_metric, mesh, inv_jacobian);
+
+    // Compute conformal Christoffel symbols.
+    const auto conformal_phi =
+        gh::phi(lapse, deriv_lapse, shift, deriv_shift, conformal_metric,
+                deriv_conformal_metric);
+    const auto conformal_christoffel_second_kind =
+        gh::christoffel_second_kind(conformal_phi, inv_conformal_metric);
+    const auto conformal_christoffel_contracted = tenex::evaluate<ti::i>(
+        conformal_christoffel_second_kind(ti::J, ti::i, ti::j));
```

Comment:
Use the Christoffel routines from `src/PointwiseFunctions/GeneralRelativity/Christoffel.hpp`, not from `GeneralizedHarmonic`